src/sentry/search/eap/resolver.py

```diff
@@ -692,24 +696,47 @@ def resolve_aggregate(
         alias = match.group("alias") or column
 
         # Get the function definition
-        if function not in self.definitions.functions:
+        function_definition: FunctionDefinition | FormulaDefinition
+        if function in self.definitions.functions:
+            function_definition = self.definitions.functions[function]
+        elif function in self.definitions.formulas:
+            function_definition = self.definitions.formulas[function]
+        else:
             raise InvalidSearchQuery(f"Unknown function {function}")
-        function_definition = self.definitions.functions[function]
 
-        parsed_columns = []
+        parsed_args: list[ResolvedColumn | Any] = []
 
         # Parse the arguments
-        attribute_args = fields.parse_arguments(function, columns)
-        if len(attribute_args) < len(function_definition.required_arguments):
+        arguments = fields.parse_arguments(function, columns)
+        if len(arguments) < len(function_definition.required_arguments):
             raise InvalidSearchQuery(
                 f"Invalid number of arguments for {function}, was expecting {len(function_definition.required_arguments)} arguments"
             )
 
         for index, argument in enumerate(function_definition.arguments):
             if argument.ignored:
                 continue
-            if index < len(attribute_args):
-                parsed_argument, _ = self.resolve_attribute(attribute_args[index])
+            if argument.validator is not None:
+                if not argument.validator(arguments[index]):
+                    raise InvalidSearchQuery(
+                        f"{arguments[index]} is not a valid argument for {function}"
+                    )
```

Comment:
nit(can be a follow up, or a TODO comment in code): can we change this so the `validator` is the one in charge of raising its error? That way we can give more specific errors?

Reply:
The validator does raise the error here, I just put this in to make it more robust. If someone mistakenly forgets to raise something, it will still get raised here. Maybe there's a way to enforce the validator raising the error?

https://github.com/getsentry/sentry/pull/85662/files#diff-8c5ca2161049ebb16b08be9488c6eec28f895ce8dce39b970a1af78bf3ead312R452